templates/elasticsearch.yaml

```diff
@@ -19,8 +19,7 @@ spec:
     - name: none
       enabled: true
   requires:
-    matchLabels:
-      kubeaddons.mesosphere.io/provides: defaultstorageclass
+    defaultStorageClass: true
```

Comment:
Since this is supposed to be a temporary workaround, we should not be adding fields to the struct that we'll have to take away. This will cause a breaking change to the API.

Reply:
We can accommodate for those changes with versioning and version to version transforms. I think this is probably one of our last opportunities to make such a change with low impact since we _still_ need to get a beta version of our API minted soon anyway.

Reply:
I do not feel that kubeaddons should be getting into determining if specific primitives are installed or configured in a specific way. This opens the gate to allowing developers to check for any object at all. Do we want to let the controller watch for a specific Event before an addon can proceed? Do we want kubeaddons to check to see if the dex-secret exists and do something different with an Addon if it does? I don't believe we do want that.

I feel kubeaddons should only act on Addon resources. It should use a driver to resolve the state of those resources. That's it. It should not be smarter than that.

Reply:
It also begins to add a level of complexity to the controller that is unnecessary. This can be solved in several other ways.

Reply:
> Do we want to let the controller watch for a specific Event before an addon can proceed? Do we want kubeaddons to check to see if the dex-secret exists and do something different with an Addon if it does? I don't believe we do want that.
> I feel kubeaddons should only act on Addon resources. It should use a driver to resolve the state of those resources. That's it. It should not be smarter than that.

I have a different take from this position. I'm for flexibility at the native Kubernetes API level. I am not opposed to enabling functionality that requires interaction with native, documented features of the Kubernetes API. We are already a Kubernetes native implementation, our API utilizes the provisions of Kubernetes and I see the default storageclass functionality as just another one of those provisions. I would be opposed to creating couplings between other APIs from other systems but not the native Kubernetes API.

Reply:
We talked about this in standup and came to alignment, updates have been made regarding that alignment.